packages/webapp/components/invite/AISearchInvite.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { ReactElement, useEffect } from 'react';
+import { Button } from '@dailydotdev/shared/src/components/buttons/Button';
+import { ProfileImageLink } from '@dailydotdev/shared/src/components/profile/ProfileImageLink';
+import KeyIcon from '@dailydotdev/shared/src/components/icons/Key';
+import { useAuthContext } from '@dailydotdev/shared/src/contexts/AuthContext';
+import { AuthTriggers } from '@dailydotdev/shared/src/lib/auth';
+import { useMutation } from 'react-query';
+import { acceptFeatureInvitation } from '@dailydotdev/shared/src/graphql/features';
+import { useRouter } from 'next/router';
+import Logo, { LogoPosition } from '@dailydotdev/shared/src/components/Logo';
+import { useFeature } from '@dailydotdev/shared/src/components/GrowthBookProvider';
+import { feature } from '@dailydotdev/shared/src/lib/featureManagement';
+import { SearchExperiment } from '@dailydotdev/shared/src/lib/featureValues';
+import { useActions } from '@dailydotdev/shared/src/hooks/useActions';
+import { ActionType } from '@dailydotdev/shared/src/graphql/actions';
+import { cloudinary } from '@dailydotdev/shared/src/lib/image';
+import { campaignConfig, JoinPageProps } from './common';
+
+export function AISearchInvite({
+  referringUser,
+  token,
+}: JoinPageProps): ReactElement {
+  const router = useRouter();
+  const search = useFeature(feature.search);
+  const { completeAction } = useActions();
+  const { user, showLogin } = useAuthContext();
+  const { mutateAsync: onAcceptMutation } = useMutation(
+    acceptFeatureInvitation,
+    {
+      onSuccess: () => {
+        completeAction(ActionType.AcceptedSearch);
```

Comment:
Can't we simply do the action in BE?
Or you want it to persist quickly in FE for checks?

(Both fine though, but should only be after accepting is applied right?

Reply:
Just wanted it to be quick since we will redirect where they will see the search bar.